src/function/table/arrow.cpp

```diff
@@ -31,6 +32,31 @@ static unique_ptr<ArrowType> CreateListType(ArrowSchema &child, ArrowVariableSiz
 
 static unique_ptr<ArrowType> GetArrowLogicalTypeNoDictionary(ArrowSchema &schema) {
 	auto format = string(schema.format);
+	// Let's first figure out if this type is a special canonical type
```

Comment:
Should we not throw an error if `!canonical_extension.empty()` and it's not something we recognize?